pkg/packagekit/updates.jsx

```diff
@@ -319,40 +319,64 @@ class ApplyUpdates extends React.Component {
                         if ("RemainingTime" in reply[0])
                             remain = reply[0].RemainingTime.v;
                         // info: see PK_STATUS_* at https://github.com/hughsie/PackageKit/blob/master/lib/packagekit-glib2/pk-enum.h
-                        this.setState({ curPackage: pfields[0] + " " + pfields[1],
-                                        curStatus: info,
+                        let newActions = this.state.actions.slice();
+                        let statusStr = PK_STATUS_STRINGS[info || PK_STATUS_ENUM_UPDATE] ||
+                            PK_STATUS_STRINGS[PK_STATUS_ENUM_UPDATE];
+                        newActions.push([statusStr, pfields[0] + " " + pfields[1]]);
+
+                        let log = document.getElementById("update-log");
+                        let atBottom = false;
+                        if (log)
+                            if (log.scrollHeight - log.clientHeight <= log.scrollTop + 2)
+                                atBottom = true;
+
+                        this.setState({ actions: newActions,
                                         percentage: percent <= 100 ? percent : 0,
                                         timeRemaining: remain > 0 ? remain : null
                         });
+
+                        // scroll update log to the bottom, if it already is (almost) at the bottom
+                        if (log && atBottom)
+                            log.scrollTop = log.scrollHeight;
                     });
             },
         });
     }
 
     render() {
-        var action;
+        var actionHTML, logRows;
 
-        if (this.state.curPackage)
-            action = (
-                <span>
-                    <strong>{ PK_STATUS_STRINGS[this.state.curStatus || PK_STATUS_ENUM_UPDATE] || PK_STATUS_STRINGS[PK_STATUS_ENUM_UPDATE] }</strong>
-                    &nbsp;{this.state.curPackage}
-                </span>
-            );
-        else
-            action = _("Initializing...");
+        if (this.state.actions.length > 0) {
+            let lastAction = this.state.actions[this.state.actions.length-1];
```

Comment:
binary operators [should be separated from operands by spaces](https://github.com/cockpit-project/cockpit/wiki/Cockpit-Coding-Guidelines#jsstyle)
`let lastAction = this.state.actions[this.state.actions.length - 1];`